mesonbuild/backend/ninjabackend.py

```diff
@@ -866,7 +866,7 @@ def create_target_linker_introspection(self, target: build.Target, linker: T.Uni
 
     def generate_target(self, target):
         try:
-            if isinstance(target, build.BuildTarget):
+            if isinstance(target, (build.BuildTarget, build.CustomTarget)):
                 os.makedirs(self.get_target_private_dir_abs(target))
         except FileExistsError:
             pass
```

Comment:
While you're here... this could be simplified to:
```python
if isinstance(target, (build.BuildTarget, build.CustomTarget)):
    os.makedirs(self.get_target_private_dir_abs(target), exist_ok=True)
```

Reply:
Glad you suggested that I do that, just pushed up that change.